hw/ip/spi_device/doc/_index.md

```diff
@@ -879,6 +879,13 @@ Due to the CDC issue, the SW is only permitted to update the registers when the
 1. The SW reads a word from TPM_CMD_ADDR CSR (optional cmdaddr_notempty interrupt).
   1. If the address falls into the return-by-HW registers and TPM_CFG.hw_reg_dis is not set, the HW does not push the command and address bytes into the TPM_CMD_ADDR CSR.
 1. The SW prepares the register value and writes the value into the read FIFO.
+  1. It is assumed that the writes are atomic.
+     TPM IP sends the read data whenever the FIFO has entries inside.
+     As TPM protocol does not support BUSY wait after sending a byte,
+     all read data should send in a chunk.
+     SW should ensure the read data is fed into the HW continuously.
+     In this version of IP, TPM provides 16B (4B wide x 4 depth) read FIFO.
+     SW may return up to 64B payload by pushing more data into the FIFO whenever the FIFO has available entries.
```

Comment:
could we specifically make a point to mention that we **may** need to disable interrupts?

Reply:
OK. Let me merge and change the doc (as backend partner needs the design asap)

Reply:
This API is quite demanding -- Disabling interrupts system-wide to effect synchronization with this one IP feels rather heavy-handed. Have you considered adding a CSR bit + hardware synchronization support instead?

Reply:
Unless TPM has 64B buffer, there's no way CSR bit helping the issue if the transfer size is bigger than 16B (this PR increases Read FIFO to 16B).

Reply:
This architecture is untenable for anything larger than the FIFO size anyway. If system software is required to disable interrupts, a larger transfer would make the critical section's timing dependent on the host's. For a 64 B transfer, system software would have to wait until the host has read 48 B before it can complete its FIFO write and re-enable interrupts. That won't fly.

Reply:
I understand the constraint is quite strict but doable. Anyhow, if you think it does not make sense to support 64B TPM read with the change, we shall increase the FIFO to 64B.

CC: @tjaychen @arnonsha

Reply:
> I understand the constraint is quite strict but doable. Anyhow, if you think it does not make sense to support 64B TPM read with the change, we shall increase the FIFO to 64B.
> 

That latter bit sounds good: Increasing the FIFO to 64 B will alleviate the problem of interrupt enablement being at the mercy of the host. However, that also means that we can easily hold off starting the transfer until a CSR write confirms the FIFO response is ready / complete. ;)

I do not precisely agree with you on the constraint, as it ignores the impact of the IP's design on the full system. We're asking system software to disable interrupts for what is generally the lowest priority class of event (bulk data transfer). The interface design here would be directly responsible for a worse system responsiveness should a user decide a different event should get low latency attention. (Though admittedly, for the use cases we've thought of for OT, that may not exist.)

Reply:
yep. That's what TPM does for Write FIFO (that's why the size has been increased to 64B). If we increase the ReadFIFO size to 64B, then we can do that too.